Source/plugins/Configuration.h

```diff
@@ -92,6 +92,7 @@ namespace Plugin {
             Core::JSON::Boolean Override;
         };
         using EnvironmentList = Core::JSON::ArrayType<Environment>;
+        static constexpr const TCHAR EnvFieldSeparator = _T(';');
```

Comment:
Good improvements! This might sound a bit like nitpicking but the idea is that the information can be used in different places (for the future) without having to parse the information. Maybe for this situation it is not really needed yet, but there are some cases ahen we might want to have access directly to the key and its value and maybe potentially the state (override yes/no).
To accomplish this I think it is good to define here a struct (mabye define it in the RPC::Object class, as I would really like to put this information in that object as well as it is part of and used by this object).
Such a class (could be a class or a struct (within the RPC::Object) and would have at least the elements, key (string), value (string) and an enum (to define a type system wide (override == true) or local (override ==false)) or a boolean... Slight preference for the enum as it is easier to extend if we need to in the future.
Than if this class/struct is part of the RPC::Object, and contained as a std::vector<RPC::Object::Environment> in this RPC::Object, I guess you do not need to extend the IShell, as the RPCObject is created in Shell.cpp (Instantiate), forwarded to the PluginServer through the IComlink interface (See: https://github.com/rdkcentral/Thunder/blob/master/Source/plugins/Shell.cpp#L85 ) and there (in the implementation of the PluginServer of Instantiate on the IComLink interface, you can applay the substitute of the %...% oin the value paty of each entry in the class, before sending them out,
So I guess that would rmean we do not have to extend the IShell interface with a *non* COMRPC compatible method. Which I guess is good news...

Reply:
updated